### ISSUE TITLE
[Refactor]: Remove `mdx` references from `tuono` npm package

### ISSUE THREAD

**Issue by marcalexiei** [tuono-labs/tuono#183]
There are some `vite` related config added in the `tuono` npm package referring to `@mdx-js/react`.
With the addition of the new `tuono.config.ts` they should be moved there rather than being in the source code of the package

**Comment by marcalexiei** [comment]
I performed a first analysis on how we can integrate the `tuono.config.ts` in order to achieve this task:

`tuono` source code affected part are the follwowings:

https://github.com/tuono-labs/tuono/blob/0839579733847fadbfe9f7d337d02809b540861f/packages/tuono/src/build/index.ts#L16-L18

and 

https://github.com/tuono-labs/tuono/blob/0839579733847fadbfe9f7d337d02809b540861f/packages/tuono/src/build/index.ts#L19-L25

### `optimizeDeps`

I think we can integrate this option directly into `vite` object without any additional logic.

### `plugins`

[From the vite doc about plugin orderdering](https://vite.dev/guide/api-plugin.html#plugin-ordering) it seems that user can control execution order using `order` and `enforce` plugin object properties but I might be missing a scenario where user needs to put specific plugins after the one we add to the config by default.

#### Solution 1

Have two separate option to define plugin that needs to be prepended

#### Solution 2

Have a single property but allow user to specify a placeholder (likely a string) than we replace with the tuono default plugins when we perform the config normalisation.

#### Solution 3

Something I haven't think of 😅

**Comment by Valerioageno** [comment]
I think a single field is enough. As far I can see, there are two situations:
- Needs to happen before -> Users use the `pre` key
- Needs to happen after -> Users use the `post` key
Aren't these two already covering all the possibilities?

**Comment by marcalexiei** [comment]
It makes sense to use a single `plugins` field then!

I preferred to ask because I haven't had to use the `vite` `enforce` plugin property yet, 
so maybe I was missing some caveats.

----

One last question before starting to setup a PR:

When using `@vitejs/plugin-react-swc`

https://github.com/tuono-labs/tuono/blob/0839579733847fadbfe9f7d337d02809b540861f/packages/tuono/src/build/index.ts#L22

I see that an `include` option is provided,
however, this option is not valid according to the [plugin documentation](https://github.com/vitejs/vite-plugin-react-swc).
I'm also unable to find the same option in the source code.

An option to achieve the same behavior seems to be [`parserConfig`](https://github.com/vitejs/vite-plugin-react-swc?tab=readme-ov-file#parserconfig) but I don't think this is required since the `include` option is not taking any effect right now so I rather going to remove it.

**Comment by Valerioageno** [comment]
Yeah. I think I run on some issues when I tried to remove it. We could try to address it in a separate PR as well!
